src/zkevm_specs/state.py

```diff
@@ -0,0 +1,540 @@
+from .util import FQ, RLC, U160, U256
+from typing import NamedTuple, Tuple, List, Sequence
+from enum import IntEnum, auto
+from .encoding import U8, is_circuit_code
+from .evm import RW, RWTableTag
+from .evm import AccountFieldTag, CallContextFieldTag
+
+MAX_KEY_DIFF = 2**32 - 1
+MAX_MEMORY_ADDRESS = 2**32 - 1
+MAX_STACK_PTR = 1023
+
+
+class Tag(IntEnum):
+    """
+    Tag used as first key in the State Circuit Rows to "select" the operation target.
+    """
+
+    Start = 1
+    Memory = 2
+    Stack = 3
+    Storage = 4
+    CallContext = 5
+    Account = 6
+    TxRefund = 7
+    TxAccessListAccount = 8
+    TxAccessListAccountStorage = 9
+    AccountDestructed = 10
+    Padding = 11
+
+
+class Row(NamedTuple):
+    """
+    State circuit row
+    """
+
+    # fmt: off
+    rw_counter: FQ
+    is_write: FQ # boolean
+    keys: Tuple[FQ, FQ, FQ, FQ, FQ] # key2 is 160bit Address.  key4 is RLC encoded
```

Comment:
I've added a little description with a link to the table.  It's hard to tell what each column is because it changes per operation, so I added the table link which contains the complete information.